### PR TITLE
NR-329534 add input windows version to reusable pre release

### PR DESCRIPTION
**Pull request by sairaj18**
Added input flag `windows_version` to reusable pre-release

### CODE REVIEW

File: .github/workflows/reusable_pre_release.yaml [hunk 1]
```diff
@@ -78,6 +78,11 @@ on:
         required: false
         type: string
         default: "infrastructure_agent/"
+      windows_version:
+        description: "The version of windows server on which build-win-packages runs"
```
Comment (rajrohanyadav) [inline .github/workflows/reusable_pre_release.yaml:82]: *nit*: I'd add something like `[Example]: 2022` in the description. Might be confused with passing `windows-2022` or `2022`. Or better, accept `windows-2022` which will make sense when checking args for the caller. `2022` lacks context.

File: .github/workflows/reusable_pre_release.yaml [hunk 2]
```diff
@@ -78,6 +78,11 @@ on:
...
+        required: false
+        type: string
+        default: "2022"
```
Comment (rajrohanyadav) [inline .github/workflows/reusable_pre_release.yaml:85]: *nit*: I'd use `windows-2022` as default as mentioned above.

File: .github/workflows/reusable_pre_release.yaml [hunk 3]
```diff
@@ -151,7 +156,7 @@ jobs:
     # related to https://github.com/actions/runner/issues/2205
     # since a success() is added by default and skipped jobs make success to fail we need to introduce this workaround
     if: ${{ always() && needs.build-artifacts.result == 'success' && inputs.run_build-win-packages }}
-    runs-on: windows-2022
+    runs-on: windows-${{ inputs.windows_version }}
```
Comment (rajrohanyadav) [inline .github/workflows/reusable_pre_release.yaml:159]: *nit*: same as above. I'd use: `runs-on: ${{ inputs.windows_version }}`

**Review by rajrohanyadav — APPROVED** [review]
LGTM. Added some nits.

File: .github/workflows/reusable_pre_release.yaml (hunk 1 above)
Comment (sairaj18) [reply .github/workflows/reusable_pre_release.yaml]: Done. Updated `2022` -> `windows-2022`

**Review by rajrohanyadav — APPROVED** [review]

**Pull request merged**